x-pack/plugins/security_solution/docs/testing/test_plans/detection_response/prebuilt_rules/installation_and_upgrade.md

```diff
@@ -815,6 +821,109 @@ And the Investigation Guide tab should NOT be displayed
 
 TODO: add scenarios https://github.com/elastic/kibana/issues/166215
 
+### Rule upgrade workflow: simple diff algorithm
+
+#### **Scenario: Rule field doesn't have an update and has no custom value**
+
+**Automation** 1 integration test
```

Comment:
Aren't these scenarios already covered nicely with unit tests from "[Adds unit tests for simple diff algorithm](https://github.com/elastic/kibana/pull/184483/files)" PR? Do you think it makes sense to add an integration-level test as well?

Reply:
It's a fair question, but I think it'll be nice to have both. The integration tests will allow us to test the entire workflow of the `upgrade/_review` endpoint and these tests are more for the possible diffing logic (which are certainly dictated by the various diffing algorithms) rather than the algorithms specifically. In these pretty straightforwards cases I think there will be a lot of overlap, but when we get to more complicated grouped fields etc, the entire process will be important to test.